nimare/transforms.py

```diff
@@ -179,10 +184,163 @@ def resolve_transforms(target, available_data, masker):
+                continue
+
+            if row.get("z"):
+                clusters = get_clusters_table(
+                    nib.funcs.squeeze_image(nib.load(row.get("z"))),
+                    self.z_threshold,
+                    self.cluster_threshold,
+                    self.two_sided,
+                    self.min_distance,
+                )
+            elif row.get("p"):
+                LGR.warning(f"No Z map for {row['id']}, using p map")
+                if self.two_sided:
+                    LGR.warning(f"Cannot use two_sided threshold using a p map for {row['id']}")
+                p_threshold = 1 - z_to_p(self.z_threshold)
+                nimg = nib.funcs.squeeze_image(nib.load(row.get("p")))
+                inv_nimg = nib.Nifti1Image(1 - nimg.get_fdata(), nimg.affine, nimg.header)
+                clusters = get_clusters_table(
+                    inv_nimg,
+                    p_threshold,
+                    self.cluster_threshold,
+                    self.min_distance,
+                )
+                # Peak stat p-values are reported as 1 - p in get_clusters_table
+                clusters["Peak Stat"] = p_to_z(1 - clusters["Peak Stat"])
+            else:
+                LGR.warning(f"No Z or p map for {row['id']}, skipping...")
+                continue
+
+            # skip entry if no clusters are found
+            if clusters.empty:
+                LGR.warning(
+                    f"No clusters were found for {row['id']} at a threshold of {self.z_threshold}"
+                )
+                continue
+
+            coordinates_dict[row["study_id"]] = {
+                "contrasts": {
+                    row["contrast_id"]: {
+                        "coords": {
+                            "space": coordinate_space,
+                            "x": list(clusters["X"]),
+                            "y": list(clusters["Y"]),
+                            "z": list(clusters["Z"]),
+                            "z_stat": list(clusters["Peak Stat"]),
+                        },
+                    }
+                }
+            }
+
+        # only the generated coordinates ('demolish')
+        coordinates_df = dict_to_coordinates(coordinates_dict, masker, space)
+
+        # merge pre-existing coordinates if they do not exist in the new dataset
+        # true for fill and replace
+        if self.merge_strategy != "demolish":
+            old_coordinates_df = dataset.coordinates[
+                ~dataset.coordinates["id"].isin(coordinates_df["id"])
+            ]
+            coordinates_df = coordinates_df.append(old_coordinates_df, ignore_index=True)
+
+        # ensure z_stat is treated as float
+        if "z_stat" in coordinates_df.columns:
+            coordinates_df["z_stat"] = coordinates_df["z_stat"].astype(float)
+
+        new_dataset = copy.deepcopy(dataset)
+        new_dataset.coordinates = coordinates_df
```

Comment:
At least for `fill` and `replace`, should the `CoordinateGenerator` also update the study metadata? Not sure if it should go in the coordinates DataFrame or the metadata one, but it seems like something that could be taken into account in some CBMAs.

Reply:
what kind of metadata are you thinking about? Like whether the coordinates are "original" or generated through nimare?

Reply:
Exactly. It could have pretty much any reasonable name, since NIMADS doesn't have anything in it for this.

Reply:
Do you have a use case for a particular CBMA in mind?

Reply:
oops ,sorry I didn't see this, yeah that makes sense!